.dockerignore

```diff
@@ -1 +1,2 @@
 Gemfile.lock
+.git/
```

Comment:
Rather than ignoring it here why dont we just copy only what's needed in the Dockerflie ? 
Change this https://github.com/twilio/twilio-ruby/blob/76aaf8c198133814fce85bd093b097e58c3b4fe0/Dockerfile#L6 to only copy src, tests, etc folder.

Reply:
Reference in go- 
https://github.com/twilio/twilio-go/blob/f8af549ad7c3f4df8785bab5748b29d7b8cbd9cd/Dockerfile#L6